Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift

```diff
@@ -50,6 +50,32 @@ public extension DeclGroupSyntax {
   }
 }
 
+
+public extension EffectSpecifiersSyntax {
+  @available(*, deprecated, message: "use throwsClause.throwsSpecifier")
+  var throwsSpecifier: TokenSyntax? {
+    throwsClause?.throwsSpecifier
+  }
+}
+
+
+public extension FunctionEffectSpecifiersSyntax {
+  @available(*, deprecated, message: "use throwsClause instead of throwsSpecifier")
+  init(
+      leadingTrivia: Trivia? = nil,
+      asyncSpecifier: TokenSyntax? = nil,
+      throwsSpecifier: TokenSyntax? = nil,
+      trailingTrivia: Trivia? = nil
```

Comment:
You should also be able to specify unexpected nodes here. This function should have the same signature as the original `FunctionEffectSpecifiersSyntax` initializer that took a `throwsSpecifier`.

Reply:
Per my other comment, API compatibility with unexpected notes is unlikely to be useful in practice. I do not think we should do this.

Reply:
I think we should do our best to maintain API compatibility wherever possible. Adding these parameters and the `unexpected` properties is very little work on our side and we shouldn’t make any assumptions on how clients are using swift-syntax. 

If you *really* disagree, the API incompatibility needs to be listed in the release notes.

Reply:
I *really* disagree. It's wasted work, and that time is better spent on more significant improvements.